### PR TITLE
Refactor/#147 refactor usercontext

### PR DESCRIPTION
**Pull request by PatrykSowinski**

### CODE REVIEW

File: src/components/UserContext.tsx [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
 import supabase from '../utils/supabase';
 
-const UserContext = React.createContext<IBasicUserInfo | null>(null);
-const UserUpdateContext = React.createContext<((newUser: IBasicUserInfo) => void) | null>(null);
-let loggedUser = supabase.auth.user() ?? null;
+const UserContext = React.createContext<IProfile | null>(null);
+const UserUpdateContext = React.createContext<((newUser: IProfile) => void) | null>(null);
+let loggedInUser = supabase.auth.user() ?? null;
 
 export const getUserInfo = async (uId: string) => {
```
Comment (wps0) [inline src/components/UserContext.tsx:9]: I personally really don't the naming of the 'uId' variable. On linux systems, the naming is uppercase and I suggest such or, eventually, a lower-case one.
```suggestion
export const getUserInfo = async (uid: string) => {
```

File: src/components/UserContext.tsx [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
+  if (!uId) {
```
Comment (wps0) [inline src/components/UserContext.tsx:10]: The same remark as above

File: src/components/Navigation.tsx [hunk 3]
```diff
@@ -5,35 +5,35 @@ import UserDropdown from './UserDropdown';
...
       setItems([{ link: '/user', label: 'User Profile' }, { link: '/user/books', label: 'User Books' }]);
     }
-  }, [userInfo]);
+  }, [user]);
 
   const logout = () => {
     if (setUser) {
       setUser({
-        id: '',
+        avtar: '',
```
Comment (wps0) [inline src/components/Navigation.tsx:30]: A typo?
```suggestion
        avatar: '',
```

File: src/components/LoginButton.tsx [hunk 4]
```diff
@@ -29,15 +29,16 @@ const LoginButton = () => {
...
-
+          const userInfo = response?.[0];
           setUser({
-            id: testUser?.id ?? '',
-            firstName: testUser?.firstName ?? 'unknown',
-            lastName: testUser?.lastName ?? 'unknown',
-            email: testUser?.email ?? 'no email',
-            avatarUrl: (await getUserAvatarURL())?.signedURL
-              ?? `${process.env.PUBLIC_URL}/image-not-found.png`,
+            avtar: (await getUserAvatarURL())?.signedURL
```
Comment (wps0) [inline src/components/LoginButton.tsx:34]: A typo?

**Review by wps0 — CHANGES_REQUESTED** [review]

File: src/components/Navigation.tsx (hunk 3 above)
Comment (PatrykSowinski) [reply src/components/Navigation.tsx:30]: no

File: src/components/LoginButton.tsx (hunk 4 above)
Comment (PatrykSowinski) [reply src/components/LoginButton.tsx:34]: no

File: src/components/UserContext.tsx (hunk 1 above)
Comment (PatrykSowinski) [reply src/components/UserContext.tsx]: Changed uId to uid.

File: src/components/Navigation.tsx (hunk 3 above)
Comment (wps0) [reply src/components/Navigation.tsx:29]: I'd still intend on naming it 'avatar` or removing the field completely, as it's no longer used anywhere, at least for now.

Comment (PatrykSowinski) [reply src/components/Navigation.tsx]: Done

**Review by wps0 — APPROVED** [review]

**Review by DLowHP — APPROVED** [review]

**Pull request merged**